### PR TITLE
Shapash report feature : use proba values in metrics

### PR DESCRIPTION
**Pull request by ThibaudReal**
# Description

This PR allows the use of proba values instead of predicted values in metrics.
To do so I had to change the way we pass the metrics. 

Now metrics is a list of dict that includes the following keys : 
- 'path' : path to the metric function (ex. : ‘sklearn.metrics.mean_squared_error’)
- 'name': name as displayed in the report (optional)
- 'use_proba_values' : bool that indicates whether or not to use proba values instead of predicted values (optional, default to False)

## Type of change

Please delete options that are not relevant.

- New feature

# How Has This Been Tested?

- pytest

**Test Configuration**:
* OS: MacOs
* Python version: 3.7
* Shapash version: 1.2.0

### CODE REVIEW

File: shapash/report/project_report.py [hunk 1]
```diff
@@ -357,10 +364,26 @@ def display_model_explainability(self):
...
         """
         Displays the performance of the model. The metrics are computed using the config dict.
-        Metrics should be given as a dict of key, value where the key is the name of the metric
-        and the value is the path (string) to a metric.
 
-        For example : config['metrics'] = {'Mean absolute error': 'sklearn.metrics.mean_absolute_error'}
+        Metrics should be given as a list of dict. Each dict contains they following keys :
+        'path' (path to the metric function, ex: 'sklearn.metrics.mean_absolute_error'),
+        'name' (optional, name of the metric as displayed in the report),
+        and 'use_proba_values' (optional, possible values are False (default) or True
```
Comment (ThomasBouche) [inline shapash/report/project_report.py:371]: Just 'proba_values' ?

File: tests/unit_tests/report/test_project_report.py [hunk 2]
```diff
@@ -186,7 +186,7 @@ def test_display_model_performance_2(self, mock_logging):
             explainer=self.xpl,
             project_info_file=os.path.join(current_path, '../../data/metadata.yaml'),
             y_test=self.df['y'],
-            config=dict(metrics={'mse': 'sklearn.metrics.mean_squared_error'})
+            config=dict(metrics=[{'path': 'sklearn.metrics.mean_squared_error'}])
         )
         report.display_model_performance()
         self.assertEqual(mock_logging.call_count, 0)
```
Comment (ThomasBouche) [inline tests/unit_tests/report/test_project_report.py:192]: Can you add test with use of 'proba_values' ?

Comment (ThibaudReal) [reply tests/unit_tests/report/test_project_report.py:192]: Indeed, thank you for noticing, I added a test

**Review by ThomasBouche — APPROVED** [review]
